tests/integration/ha/test_backups.py

```diff
@@ -0,0 +1,568 @@
+
+
+@pytest.mark.parametrize(
+    "cloud_name",
+    [
+        (pytest.param("microceph", marks=pytest.mark.group("microceph"))),
+        (pytest.param("aws", marks=pytest.mark.group("aws"))),
+    ],
+)
+@pytest.mark.abort_on_fail
+async def test_restore_cluster(
+    ops_test: OpsTest, cloud_configs, cloud_credentials, cloud_name
+) -> None:
+    """Restores the cluster and tries to search for index."""
+    global global_backup_id
+
+    unit_ip = await get_leader_unit_ip(ops_test)
+    app = (await app_name(ops_test)) or APP_NAME
+    leader_id = await get_leader_unit_id(ops_test)
+    config = cloud_configs[cloud_name]
+
+    logger.info(f"Syncing credentials for {cloud_name}")
+    await _configure_s3(ops_test, config, cloud_credentials[cloud_name], app)
+
+    logger.info("Restoring backup")
+    assert await restore_cluster(
+        ops_test,
+        global_backup_id,  # backup_id
+        unit_ip,
+        leader_id,
+    )
+    assert await start_and_check_continuous_writes(ops_test, unit_ip, app)
+
+
+@pytest.mark.parametrize(
+    "cloud_name",
+    [
+        (pytest.param("microceph", marks=pytest.mark.group("microceph"))),
+        (pytest.param("aws", marks=pytest.mark.group("aws"))),
+    ],
+)
+@pytest.mark.abort_on_fail
+async def test_restore_cluster_after_app_destroyed(
+    ops_test: OpsTest, cloud_configs, cloud_credentials, cloud_name
+) -> None:
+    """Deletes the entire OpenSearch cluster and redeploys from scratch.
+
+    Restores the backup and then checks if the same TEST_BACKUP_INDEX is there.
+    """
+    global global_backup_id
+
+    app = (await app_name(ops_test)) or APP_NAME
+
+    logging.info("Destroying the application")
+    await ops_test.model.remove_application(app, block_until_done=True)
+    app_num_units = 3
+    my_charm = await ops_test.build_charm(".")
+    config = cloud_configs[cloud_name]
+
+    # Redeploy
+    await asyncio.gather(
+        ops_test.model.deploy(my_charm, num_units=app_num_units, series=SERIES),
+    )
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.integrate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.integrate(APP_NAME, S3_INTEGRATOR)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME],
+        status="active",
+        timeout=1400,
+        idle_period=IDLE_PERIOD,
+    )
+
+    leader_id = await get_leader_unit_id(ops_test)
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    logger.info(f"Syncing credentials for {cloud_name}")
+    await _configure_s3(ops_test, config, cloud_credentials[cloud_name], app)
+
+    logger.info("Restoring backup")
+    assert await restore_cluster(
+        ops_test,
+        global_backup_id,  # backup_id
+        leader_unit_ip,
+        leader_id,
+    )
+
+    logger.info("Creating backup")
+    backup_id = await backup_cluster(
+        ops_test,
+        leader_id,
+    )
+    assert backup_id > 0
+    if cloud_name not in backups_by_cloud:
```

Comment:
we need count the restored documents and compare them with the original count of the original cluster